assertion/function/assertiontree/backprop.go

```diff
@@ -578,7 +578,18 @@ buildShadowMask:
 					}
 
 					lhsNode, ok := rootNode.LiftFromPath(lpath)
-					if ok {
+					// TODO: below check for `lhsNode != nil` should not be needed when NilAway supports Ok form for
+					//  used-defined functions (tracked issue #77)
```

Comment:
👍 

(BTW this function is long enough to the point it's a bit painful to (re-)read every time there is a change to it, we probably want to refactor it to smaller pieces in future revisions).

Reply:
Oh yes, agreed. This can definitely benefit from refactoring :)